sys/arduino/SPI.cpp

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2019 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     sys_arduino_api
+ * @{
+ *
+ * @file
+ * @brief       Implementation of the Arduino 'SPI' interface
+ *
+ * @author      Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+ *
+ * @}
+ */
+
+extern "C" {
```

Comment:
Compiling
```
make BOARD=arduino-mega2560 -C examples/arduino_hello-world/
```
fails even though it doesn't use `SPI.h`. The reason is that `SPI.cpp` which includes `SPI.h` is compiled when `sys/arduino` is compiled. The code in `SPI.h` has to be embedded in `#if MODULE_PERIPH_SPI ... #endif`.

Reply:
I solved it slightly differently: I guarded the implementation in `SPI.cpp` to be only provided if SPI support is present. This allows my to keep the more human readable error message during compilation. (Otherwise during linking an unresolved symbol error would come up, which looks more like a bug in the RIOT implementation than missing SPI support for a board.)